pkg/storage/storage.go

```diff
@@ -21,28 +24,31 @@ type ServiceStorage interface {
 	DeleteNamespace(namespace string) error
 }
 
-// NewStorage creates a new storage provider based on the input
-func NewStorage(storageProvider Storage) (ServiceStorage, error) {
-	switch storageProvider {
-	case Bolt:
-		return NewBoltDB()
-	default:
-		return nil, fmt.Errorf("unsupported storage provider: %s", storageProvider)
+// NewStorage creates a new storage provider and provider specify param based on the input
+func NewStorage(storageProvider Storage, param interface{}) (ServiceStorage, error) {
+	impl := GetStorage(storageProvider)
+	if impl == nil {
+		return impl, fmt.Errorf("unsupported storage provider: %s", storageProvider)
 	}
+	err := impl.Init(param)
+	return impl, err
 }
 
-// AvailableStorage returns the supported storage providers
-func AvailableStorage() []Storage {
```

Comment:
I think this method is still useful for "default" storage providers

Reply:
if we have "default"  storage setting in configure, this will not be a problem. but i think will still need `AvailableStorage ` for example, show a list of available storages.  so keep it ?

Reply:
yes